op-nat/runner/runner.go

```diff
@@ -0,0 +1,660 @@
+package runner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/ethereum-optimism/infra/op-nat/registry"
+	"github.com/ethereum-optimism/infra/op-nat/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// TestResult captures the outcome of a single test run
+type TestResult struct {
```

Comment:
should we have some support for test artifacts here? logs for  example

Reply:
Yes, but I was thinking of tackling this in a separate PR. May be non-trivial to parse it nicely.
https://github.com/ethereum-optimism/infra/issues/198 created to capture the work.

Sound OK?